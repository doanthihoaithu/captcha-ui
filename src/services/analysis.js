import { requestServices } from 'services';
import * as constants from 'constants/index'


const analyzeQuery = (accessToken, data) => requestServices.customAxios.post(
    constants.analysis_analyze_query,
    data,
    {
        headers: {
            'Authorization': `Bearer ${accessToken}`,
            'Content-Type': 'multipart/form-data'
        }
    }
)

const generateQuery = (data) => requestServices.customAxios.post(
    constants.generate_captcha_query,
    data,
    {
        headers: {
            // 'Authorization': `Bearer ${accessToken}`,
            'Content-Type': 'multipart/form-data',
            // 'Content-Type': 'application/json'
        }
    }
)

export default {
    analyzeQuery,
    generateQuery
}