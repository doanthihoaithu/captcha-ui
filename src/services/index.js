// @flow
export { default as userServices } from './users';
export { default as requestServices } from './requests';
export { default as rootReducer } from './rootReducer';
export { default as store } from './store'
