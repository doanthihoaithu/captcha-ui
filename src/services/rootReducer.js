import { combineReducers } from 'redux';
import userReducer from './redux/user/reducers';
import statusReducer from './redux/status/reducers';

export default combineReducers({
	user: userReducer,
	status: statusReducer
});