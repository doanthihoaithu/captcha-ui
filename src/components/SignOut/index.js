import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { userServices } from 'services';
import * as actions from "../../services/redux/user/actions";
import animationTrigger from 'helpers/loadingAnimationTrigger';
import { toast } from 'react-toastify';

function SignOut(props) {
    const dispatch = useDispatch();
    const accessToken = useSelector(state => state.user.accessToken);

    function signUserOut() {
        animationTrigger.startLoading();
        userServices.signUserOut(accessToken).then(res => {
            // console.log(res);
            animationTrigger.stopLoading();
            dispatch(actions.logOut());
            toast.success("Logout successful");
        }).catch(err => {
            // console.log(err);
            animationTrigger.stopLoading();
        })
    }

    return (
        <button className="btn btn-danger mx-5 home-btn" onClick={signUserOut}>Sign Out</button>
    )
}

export default SignOut