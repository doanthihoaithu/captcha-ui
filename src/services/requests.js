// @flow
import axios from 'axios';
import * as actions from 'services/redux/user/actions';
import { store } from 'services'
import { toast } from 'react-toastify'

export const BASE_API_URL: string = `${process.env.REACT_APP_BASE_API_URL || 'localhost:3000'}`;

const customAxios = axios.create({
  baseURL: BASE_API_URL,
});

customAxios.interceptors.response.use(
  response => response,
  error => {
    // Add logic to handle error here
    // console.log(error.response.data)
    if (error.response && error.response.data.code === 500) {
      switch (error.response.data.message) {
        case "Signature has expired":
          store.dispatch(actions.logOut());
          toast.error('Your session has expired. Please sign in to continue');
          return Promise.reject(error);
        default:
          return Promise.reject(error);
      }
    }

    return Promise.reject(error);
  },
);

export default { customAxios };
