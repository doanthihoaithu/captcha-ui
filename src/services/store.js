import { createStore } from 'redux';
import { rootReducer } from './../services';

const store = createStore(rootReducer);

export default store;
