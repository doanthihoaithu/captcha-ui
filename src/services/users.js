// @flow
import { requestServices } from 'services';
import * as constants from 'constants/index'

// const fetchUsers = () => requestServices.customAxios.get('/').then(res => res.data);

const signUserIn = (data) => requestServices.customAxios.post(
    constants.user_authenticate_address,
    data,
    {
        header: {
            'Content-Type': 'application/json',
            // 'Access-Control-Allow-Origin': '*',
        }
    }
)

const signUserOut = (accessToken) => requestServices.customAxios.post(
    constants.user_sign_out_address,
    {},
    {
        headers: {
            'Authorization': `Bearer ${accessToken}`
        }
    }
)

const registerNewUser = (data) => requestServices.customAxios.post(
    constants.user_sign_up_address,
    data,
    {
        headers: {
            'Content-Type': 'application/json',
            // 'Access-Control-Allow-Origin': '*'
        }
    }
)

const resetPassword = (data) => requestServices.customAxios.post(
    constants.user_forget_password_address,
    data,
    {
        headers: {
            'Content-Type': 'application/json',
            // 'Access-Control-Allow-Origin': '*'
        }
    }
)

const changePassword = (data, accessToken) => requestServices.customAxios.post(
    constants.user_change_password_address,
    data,
    {
        headers: {
            'Content-Type': 'application/json',
            'Authorization': `Bearer ${accessToken}`
            // 'Access-Control-Allow-Origin': '*'
        }
    }
)

const updateAvatar = (username, data, accessToken) => requestServices.customAxios.post(
    constants.user_update_avatar_address + username,
    data,
    {
        headers: {
            'Content-Type': 'multipart/form-data',
            'Authorization': `Bearer ${accessToken}`
        }
    }
)

const getAvatar = (username) => requestServices.customAxios.get(
    constants.user_get_avatar_address + username,
)

export default {
    signUserIn,
    signUserOut,
    registerNewUser,
    resetPassword,
    changePassword,
    updateAvatar,
    getAvatar,
};
