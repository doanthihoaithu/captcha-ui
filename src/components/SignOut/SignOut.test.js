import React from 'react'
import { shallow, render, mount } from 'enzyme'
import { Provider } from 'react-redux'
import { store, userServices } from './../../services'

import SignOut from '.'

let component;

beforeEach(() => {
    component = mount(
        <Provider store={store}>
            <SignOut />
        </Provider>
    )
})

it('should match snapshot', () => {
    expect(component).toMatchSnapshot()
})


it('should call function signUserOut when being clicked', () => {
    const signOutBtn = component.find("button");
    expect(signOutBtn.text()).toBe("Sign Out")

    const signUserOutFn = jest.spyOn(userServices, 'signUserOut')

    signOutBtn.simulate("click")

    expect(signUserOutFn).toHaveBeenCalledTimes(1)


})