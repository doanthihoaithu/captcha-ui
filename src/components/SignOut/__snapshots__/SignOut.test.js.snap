// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <SignOut>
    <button
      className="btn btn-danger mx-5 home-btn"
      onClick={[Function]}
    >
      Sign Out
    </button>
  </SignOut>
</Provider>
`;
